__tests__/units/observer.spec.ts

```diff
@@ -0,0 +1,604 @@
+import { EventObserver } from '../../src/observers';
+import { getRandom } from '../util/fn';
+let observer: EventObserver;
+describe('Observer', () => {
+  beforeEach(() => {
+    observer = new EventObserver(window);
+  });
+  it('Create an instance', () => {
+    expect(observer).not.toBeNull();
+  });
+
+  it('should handle state', () => {
+    // the initial state should be inactive;
+    expect(observer['state']).toBe('inactive');
+
+    // inactive -> active: active
+    observer.start();
+    expect(observer['state']).toBe('active');
+
+    // active -> suspend: suspend
+    observer.suspend();
+    expect(observer['state']).toBe('suspend');
+
+    // suspend -> active: active;
+    observer.start();
+    expect(observer['state']).toBe('active');
+
+    // active -> inactive: inactive
+    observer.stop();
+    expect(observer['state']).toBe('inactive');
+
+    // inactive -> suspend: inactive
+    observer.suspend();
+    expect(observer['state']).toBe('inactive');
+
+    // suspend -> inactive: inactive
+    observer.start();
+    observer.suspend();
+    observer.stop();
+    expect(observer['state']).toBe('inactive');
+  });
+
+  describe('event handler', () => {
+    let emitSpy: jasmine.Spy;
+    beforeAll(() => {
+      jest.useFakeTimers();
+    });
+
+    beforeEach(() => {
+      // throttle manager doesn't work properly in jest dom envrionment, mock it.
+      spyOn(observer['throttleManager'], 'getThrottle').and.callFake(
+        (_divider, func: (...args: any[]) => void) => {
+          return function (this: any, ...args: any[]) {
+            _divider instanceof Function
+              ? _divider.apply(this, args)
+              : _divider;
+            func.apply(this, args);
+          };
+        },
+      );
+      observer.start();
+      observer.suspend();
+      emitSpy = spyOn<any>(observer, 'onEmit');
+    });
+
+    afterEach(() => {
+      emitSpy.calls.reset();
+    });
+
+    describe('observe mouse interaction', () => {
```

Comment:
should we need test every event handler?
maybe we just need to test the functions, but event observer should be test in puppeteer